### DISCUSSION TITLE
An addition to the Vector types (rotate_toward)

### DISCUSSION THREAD

**Discussion by OllyMan21** [godotengine/godot-proposals#7302] [Engine Core]
I propose a combination of aspects from the move_toward and slerp methods of the Vector types.

Instead of using a weight for slerp we use a "delta" in such a way that the vector to be rotated is rotated by an angle corresponding to "delta". Much like how move_toward moves the vector (A) towards vector (B) along A.direction_to(B) with a magnitude of delta instead of a weight like in lerp, we replace the slerp weight with a delta and the length of the vector is adjusted the same way slerp would.

**Comment by theraot** [comment]
See https://github.com/godotengine/godot-proposals/issues/2074 (it is not the same thing).

**Reply by OllyMan21** [reply]
Thank you. That is part of the way to solving the "issue". However,  I can't help but feel that would be an inconsistent naming convention since move_toward returns a vector and their rotate_toward would return an angle and also requires 3 parameters. I like their idea but I personally feel that my mechanism suits the name "rotate_toward" better (or vice versa). I like to see my proposal as an analogue to the relationship between lerp and move_toward. That is, move_toward is to lerp as rotate_toward is to slerp.

**Reply by theraot** [reply]
@OllyMan21 I want to bring to your attention that there is another `move_toward`: https://docs.godotengine.org/en/stable/classes/class_@globalscope.html#class-globalscope-method-move-toward
